git-secrets

```diff
@@ -68,8 +80,14 @@ scan() {
   process_output $? "${output}"
 }
 
+# Scans through history using patterns
 scan_history() {
-  local to_scan=$(git rev-list --all)
+  # git log does not support multiple patterns, so we need to combine them
+  local combined_patterns=$(load_combined_patterns)
+  [ -z "${combined_patterns}" ] && return 0
+  # Looks for differences matching the patterns, reduces the number of revisions to scan
+  local to_scan=$(git log --all -G"${combined_patterns}" --pretty=%H)
+  # Scan through revisions with findings to normalize output
   local output=$(git_grep "" "${to_scan}")
```

Comment:
I know this isn't part of your PR, but I think there might be a bug on this line: when you use `local`, bash counts it as a function call and `$?` is assigned the result of invoking local (it doesn't pick up the failure in the underlying call). As this PR is superseding the other can PRs, could you also update this line?

Here's my test script to show local overriding the value of `$?`:

```bash
#!/bin/sh

foo() {
  local a=$(ls ~/not-there.dat)
  echo $?
  ls ~/not-there.dat
  echo $?
}

foo
```